_data-prepper/pipelines/configuration/sources/s3.md

```diff
@@ -43,25 +84,70 @@ Option | Required | Type | Description
+expression_type | No  | String | The type of the provided expression. The default value is `SQL`. Maps directly to the [ExpressionType](https://docs.aws.amazon.com/AmazonS3/latest/API/API_SelectObjectContent.html#AmazonS3-SelectObjectContent-request-ExpressionType)
+input_serialization | Yes, when using `s3_select` | String  | Provides the S3 Select file format. Amazon S3 uses this format to parse object data into records and returns only records that match the specified SQL expression. Maybe be `csv`, `json`, or `parquet`.
+compression_type | No | String  | Specifies object's compression format. Maps directly to the [CompressionType](https://docs.aws.amazon.com/AmazonS3/latest/API/API_InputSerialization.html#AmazonS3-Type-InputSerialization-CompressionType)
+csv | No | [csv](#s3_select_csv) | Provides the CSV configuration for processing CSV data.
+json | No | [json](#s3_select_json) | Provides the JSON configuration for processing JSON data.
+
+### csv<a name="s3_select_csv"></a>
+
+Use the following options in conjunction with the `csv` configuration for `s3_select`  to determine how your parsed CSV file should be formatted.
+
+These options map directly to options available in S3 Select's [CSVInput](https://docs.aws.amazon.com/AmazonS3/latest/API/API_CSVInput.html) data type.
+
+Option | Required | Type        | Description
+:--- |:---------|:------------| :---
+file_header_info | No  | String  | Describes the first line of input. Maps directly to the [FileHeaderInfo](https://docs.aws.amazon.com/AmazonS3/latest/API/API_CSVInput.html#AmazonS3-Type-CSVInput-FileHeaderInfo) property.
+quote_escape | No  | String  | A single character used for escaping the quotation mark character inside an already escaped value. Maps directly to the [QuoteEscapeCharacter](https://docs.aws.amazon.com/AmazonS3/latest/API/API_CSVInput.html#AmazonS3-Type-CSVInput-QuoteEscapeCharacter) property.
+comments | No   | String  | A single character used to indicate that a row should be ignored when the character is present at the start of that row. Maps directly to the [Comments](https://docs.aws.amazon.com/AmazonS3/latest/API/API_CSVInput.html#AmazonS3-Type-CSVInput-Comments) property.
+
+#### json<a name="s3_select_json"></a>
+
+Use the following option in conjunction with the `json` for `s3_select` to how S3 Select processes the JSON file.
+
+Option | Required | Type  | Description
+:--- |:---------|:------------| :---
+type | No  | String  | The type of JSON. May be either `DOCUMENT` or `LINES`. Maps directly to the [Type](https://docs.aws.amazon.com/AmazonS3/latest/API/API_JSONInput.html#AmazonS3-Type-JSONInput-Type) property.
 
-Option | Required | Type | Description
-:--- | :--- | :--- | :---
-name | Yes | String | Name of the pipeline to read from.
 
 ## Metrics
```

Comment:
In other docs, we have this section formatted as a table (metric name/type/description). The type of metric should definitely be included here, especially if it's a custom metric specific to source.

Reply:
I'll reconcile these in the great processor reconciliation PR I'm working on.